concrete/src/Job/JobServiceProvider.php

```diff
@@ -0,0 +1,20 @@
+<?php
+namespace Concrete\Core\Job;
+
+use Concrete\Core\Foundation\Service\Provider as ServiceProvider;
+
+class JobServiceProvider extends ServiceProvider
+{
+    public function register()
+    {
+        // Supply the deprecated static session accessor with a real application object
+        Job::setApplicationObject($this->app);
+
+        $this->app->bind('Concrete\Core\Job\Factory', 'Concrete\Core\Job\Factory');
+        $this->app->bind('Concrete\Core\Job\Service', 'Concrete\Core\Job\Service');
+
+        $this->app->singleton('job', function ($app) {
```

Comment:
I think what you are trying to do here is this:

```php
$this->app->bind('job', Factory::class);
$this->app->singleton(Factory::class);
```

However I don't see any reason for this class to be a singleton. You should only need `$this->app->bind('job', Factory::class);`